service/controller/resource/remotewritesecret/create.go

```diff
@@ -0,0 +1,59 @@
+package remotewritesecret
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/resourcecanceledcontext"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.Debugf(ctx, "ensuring prometheus remotewrite secret")
+	{
+		remoteWrite, err := ToRemoteWrite(obj)
```

Comment:
I think the `ToRemoteWrite` function could be moved into a different packages and be used here and in the `prometheusRemoteWrite` resource also.